pandas/core/indexes/base.py

```diff
@@ -2195,35 +2195,21 @@ def union(self, other):
                 value_set = set(self._values)
                 result.extend([x for x in other._values if x not in value_set])
         else:
-            indexer = self.get_indexer(other)
-            indexer, = (indexer == -1).nonzero()
-
+            indexer = np.where(self.get_indexer(other) == -1)[0]
             if len(indexer) > 0:
                 other_diff = algos.take_nd(other._values, indexer,
                                            allow_fill=False)
                 result = _concat._concat_compat((self._values, other_diff))
 
-                try:
-                    self._values[0] < other_diff[0]
-                except TypeError as e:
-                    warnings.warn("%s, sort order is undefined for "
-                                  "incomparable objects" % e, RuntimeWarning,
-                                  stacklevel=3)
-                else:
-                    types = frozenset((self.inferred_type,
-                                       other.inferred_type))
-                    if not types & _unsortable_types:
-                        result.sort()
-
             else:
                 result = self._values
 
-                try:
-                    result = np.sort(result)
-                except TypeError as e:
-                    warnings.warn("%s, sort order is undefined for "
-                                  "incomparable objects" % e, RuntimeWarning,
-                                  stacklevel=3)
+            try:
```

Comment:
you probably can simply use ``core.sorting.safe_sort`` here instead and just remove the warning entirely (as it fails in many less cases)

Reply:
Done, but I left the warning as it's still raised if not even ``safe_sort`` can sort them.

(I assume the failure is a problem with AppVeyor)